iot-hub/Samples/device/EdgeDownstreamDevice/Program.cs

```diff
@@ -0,0 +1,131 @@
+        /// Next attempt to connect to the Edge device and send it DEVICE_MESSAGE_COUNT
+        /// number of telemetry data messages.
+        ///
+        /// Note: Either set the DEVICE_MESSAGE_COUNT environment variable with the number of
+        /// messages to be sent to the IoT Edge runtime or set it in the launchSettings.json.
+        /// </summary>
+        static void Main()
+        {
+            InstallCACert();
+
+            try
+            {
+                MESSAGE_COUNT = Int32.Parse(Environment.GetEnvironmentVariable("DEVICE_MESSAGE_COUNT"), NumberStyles.None, new CultureInfo("en-US"));
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Invalid number of messages in env variable DEVICE_MESSAGE_COUNT. MESSAGE_COUNT set to {0}\n", MESSAGE_COUNT);
+            }
+
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(DeviceConnectionString);
+
+            if (deviceClient == null)
+            {
+                Console.WriteLine("Failed to create DeviceClient!");
+            }
+            else
+            {
+                SendEvents(deviceClient).Wait();
+            }
+
+            Console.WriteLine("Exiting!\n");
+        }
+
+        /// <summary>
+        /// Add certificate in local cert store for use by downstream device
+        /// client for secure connection to IoT Edge runtime.
+        ///
+        ///    Note: On Windows machines, if you have not run this from an Administrator prompt,
+        ///    a prompt will likely come up to confirm the installation of the certificate.
+        ///    This usually happens the first time a certificate will be installed.
+        /// </summary>
+        static void InstallCACert()
+        {
+            string certPath = Environment.GetEnvironmentVariable("EDGE_DEVICE_CA_CERTIFICATE_PATH");
+            if (!string.IsNullOrWhiteSpace(certPath))
+            {
+                Console.WriteLine("User configured CA certificate path: {0}", certPath);
+                if (!File.Exists(certPath))
+                {
+                    // cannot proceed further without a proper cert file
+                    Console.WriteLine("Invalid certificate file: {0}", certPath);
+                    throw new InvalidOperationException("Invalid certificate file.");
+                }
+                else
+                {
+                    Console.WriteLine("Attempting to install CA certificate: {0}", certPath);
+                    X509Store store = new X509Store(StoreName.Root, StoreLocation.CurrentUser);
+                    store.Open(OpenFlags.ReadWrite);
+                    store.Add(new X509Certificate2(X509Certificate2.CreateFromCertFile(certPath)));
+                    Console.WriteLine("Successfully added certificate: {0}", certPath);
+                    store.Close();
+                }
+            }
+            else
```

Comment:
Add the branch at the top and decrease indentation to the rest of the code please instead of having an if/else this requires only an if (...) { [...] return or throw; }